docs/USAGE-GUIDE.md

```diff
@@ -1777,6 +1777,216 @@ handle. The handle is considered invalid after **s2n_connection_free** is used.
 [s2n_connection_wipe](#s2n\_connection\_wipe) does not need to be called prior to this function. **s2n_connection_free** performs its own wipe
 of sensitive data.
 
+## TLS1.3 Pre-Shared Key Related Calls
+
+S2N-TLS provides support for pre-shared keys (PSKs). PSKs can be established out of band (external PSKs) or in a previous connection (resumption PSKs) for versions >= TLS1.3.
+The PSK is used to mutually authenticate the peers, under the assumption that it is not shared with others.
+This allows the handshake to skip the certificate verification steps, saving bandwidth and latency.
+
+### Benefits of Using Pre-Shared Keys
+
+- Using pre-shared keys can avoid the need for public key operations.
+  This is useful in performance-constrained environments with limited CPU power.
+- PSKs may be more convenient from a key management point of view. For instance, in closed environments where the connections
+  are configured manually in advance, it is easier to configure a PSK than to use certificates. Another case is when the parties already have a mechanism for setting up a shared secret key, and that mechanism could be used to "bootstrap" a key for authenticating the TLS connection.
+- PSKs are more appropriate for use cases with long-lived credentials stored in hardware secure elements than PKI-based
+  solutions which enforces expiration dates and requires error-prone rotation schemes.
+
+### Security Considerations
+
```

Comment:
I prefer:
- A peer who acts as both server and client _should not_ use the same PSK for both roles. Doing so would open up the handshake to a selfie attack. This attack is elaborated in greater detail here: [Selfie: reflections on TLS 1.3 with PSK](https://eprint.iacr.org/2019/347.pdf).

Reply:
This suggestion is insufficient as it misses that `each endpoint must use a different PSK`. 
and the use of `who` and `open up the..` reads more colloquial that written language.


```
A peer that acts as both server and client should not use the same PSK for both roles. 
Similarly, the peer should not use the same PSK on different endpoints. Doing so would expose 
the handshake to a selfie attack. This attack is elaborated in detail here: 
[Selfie: reflections on TLS 1.3 with PSK](https://eprint.iacr.org/2019/347.pdf) 
```